### PR TITLE
rewrite self state monitor

### PR DESCRIPTION
**Pull request by AlexAkulov**

### CODE REVIEW

**Comment by gmlexx** [comment]
What about code coverage decrease? There is a lot of code that need to be tested.
@AlexAkulov who will write the tests? Me or you?

File: selfstate.go [hunk 1]
```diff
@@ -6,9 +6,9 @@ import (
 	"time"
 )
 
-// CheckSelfStateMonitorSettings -
+// CheckSelfStateMonitorSettings - validate contact types
 func CheckSelfStateMonitorSettings() error {
-	if config.Notifier.SelfState.Enabled != "true" {
+	if ToBool(config.Notifier.SelfState.Enabled) {
 		return nil
```
Comment (gmlexx) [inline selfstate.go:15]: It seems to bug.
If self check enabled, we exit from function and do not check contact settings

**Comment by gmlexx** [comment]
Starting to write tests and found a bug. Look at comment ^

**Comment by AlexAkulov** [comment]
Yes, it is bug

**Pull request merged**